packages/wrangler/src/event-subscriptions/index.ts

```diff
@@ -0,0 +1,323 @@
+		);
+
+		logger.table([
+			{
+				id: subscription.id,
+				name: subscription.name,
+				source: `${subscription.product_id}.${subscription.resource_id}`,
+				destination: `queues.${subscription.queue_name}`,
+				events: subscription.events.join(", "),
+			},
+		]);
+	},
+});
+
+export const eventSubscriptionsCreateCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description:
+			"Create a new Event Subscription. Use `event-subscriptions browse` to see available services and events.",
+	},
+
+	positionalArgs: ["name"],
+	args: {
+		name: {
+			type: "string",
+			description: "",
+			demandOption: true,
+		},
+		source: {
+			type: "string",
+			description: "<Service>.<Resource> to subscribe to",
+			demandOption: true,
+		},
+		destination: {
+			type: "string",
+			description: "Name of the Queue to send events to",
+			demandOption: true,
+		},
+		events: {
+			type: "string",
+			description: "Names of events to subscribe to",
+			demandOption: true,
+		},
+	},
+	validateArgs({ source }) {
+		const [productId, resourceId] = source.split(".");
+
+		const product = PRODUCTS.find((p) => p.id === productId);
+		if (!product) {
+			throw new CommandLineArgsError(
+				`Invalid source. TODO: Write a better / more helpful error message`
+			);
+		}
+
+		if (product.resource) {
+			if (!resourceId) {
+				throw new CommandLineArgsError(
+					`Invalid resource ID. TODO: Write a better / more helpful error message`
+				);
+			}
+		}
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const { name, source, destination: queueId, events } = args;
+
+		const [productId, resourceId] = source.split(".");
+		const product = PRODUCTS.find((p) => p.id === productId);
+
+		// Disabling ESLint and using `!` because we've validated the argument already
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const res = await product!.subscription.create(ctx, {
+			accountId,
+			name,
+			resourceId,
+			queueId,
+			events: events.split(",").map((e) => e.trim()),
+		});
+
+		logger.table([
+			{
+				id: res.id,
+				name,
+				source,
+				destination: `queues.${queueId}`,
+				events,
+			},
+		]);
+	},
+});
+
+export const eventSubscriptionsUpdateCommand = createCommand({
+	metadata: {
+		owner: "Product: Queues",
+		status: "open-beta",
+		description: "Update an event subscription",
+	},
+
+	positionalArgs: ["id"],
+	args: {
+		id: {
+			type: "string",
+			description: "ID of the event subscription",
+			demandOption: true,
+		},
+		name: {
+			type: "string",
+			description: "Name to set",
+		},
+		events: {
+			type: "string",
+			description: "Names of events to subscribe to",
+		},
+	},
+
+	async handler(args, ctx) {
+		const config = readConfig(args);
+		const accountId = await requireAuth(config);
+
+		const { id, name, events } = args;
+
+		const subscription = await ctx.fetchResult<EventSubscription>(
```

Comment:
what will the API return if subscription with `id` does not exist? Should this case be handled here?

Reply:
Same as above, will add a test!